content/docs/guides/build/manual-llvm.md

```diff
@@ -0,0 +1,103 @@
+---
+title: "Manual LLVM build"
+weight: 3
+description: >
+  How to build LLVM yourself, or "manually".
+---
+
+
+| You need to build LLVM manually in the following cases |
+|---|
+|  You would like to use it for the ESP8266 or ESP32 chips |
+| You are using Windows. |
+| Your Linux distribution (if you use Linux) does not ship the right LLVM version. |
+
+
+## Build dependencies
+***Depending on which OS you use the command to install
+the dependencies needed to build LLVM will differ.*** Skip over to the command for your Operating System (OS) in this section (OS names are bolded):
+
+
+**Linux Debian and Ubuntu** users may install all required tools this way:
+```shell
+sudo apt-get install build-essential git cmake ninja-build
+```
+
+---
+
+**Linux Fedora** users may install all required tools with:
+```shell
+sudo dnf groupinstall "Development Tools"
+sudo dnf install cmake ninja-build
+```
+
+---
+
+**MacOS** users may install these tools using [Homebrew](https://brew.sh/):
+
+```shell
+brew install cmake ninja
+```
+
+---
+
+**Windows** users may install build dependencies using [Chocolatey](https://chocolatey.org/). Install Chocolatey first, then run the following in a command prompt or PowerShell with administrative privileges:
+
+```shell
+choco install --confirm git golang mingw make cmake ninja python
+```
+**Windows** users should also use *Git Bash* (installed above) to run all the build commands like `make`. The TinyGo build system expects a Unix-like environment that is not normally provided by Windows but is included already in *Git Bash*.
+
+Choco doesn't seem to add CMake automatically to the `$PATH` variable. You can do this manually if needed, in Git bash:
+
+```shell
+export PATH="$PATH:/c/Program Files/CMake/bin"
+```
+
+## Building LLVM
+***The following instructions are common to all operating systems.***
+
+Background: *[LLVM](https://llvm.org/) is used by TinyGo to create smaller programs than can be compiled with normal Go distribution. It is commonly referred to as a *compiler infrastructure* since it provides several tools to build, link and optimize programs which adhere to its framework.*
```

Comment:
Almost, but not exactly. LLVM is a compiler toolkit. It's not _the_ reason why TinyGo produces such small binaries (in principle the upstream Go version could do that as well, if they think it's important enough).
What about this?
> LLVM is a library and collection of tools for building compilers. It is the base of many newer compilers, like Clang, Swift, and the Rust compiler so they can all build on the same high-quality base. Similarly, TinyGo uses LLVM for all its low-level optimization and machine code generation so that it can produce code that's roughly as small and efficient as all these other modern compilers.

(Perhaps a bit verbose. Just wanted to cut out the pat that says LLVM reduces binary size, which is mostly but not strictly true).